src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.engine.dataflow;
+
+import com.google.api.services.genomics.model.Read;
+import com.google.api.services.storage.Storage;
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.options.PipelineOptionsFactory;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.genomics.dataflow.readers.bam.BAMIO;
+import com.google.cloud.genomics.dataflow.readers.bam.ReadBAMTransform;
+import com.google.cloud.genomics.dataflow.utils.GCSOptions;
+import com.google.cloud.genomics.dataflow.utils.GenomicsOptions;
+import com.google.cloud.genomics.utils.Contig;
+import com.google.cloud.genomics.utils.GenomicsFactory;
+import com.google.common.collect.ImmutableList;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+import org.broadinstitute.hellbender.utils.dataflow.DataflowUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Class to load reads into a PCollection from either a cloud storage bucket or a local bam file.
+ */
+public class ReadsSource {
+    private final String bam;
+    private final boolean cloudStorageUrl;
+    private GCSOptions options;
+    private GenomicsFactory.OfflineAuth auth;
+
+    public ReadsSource(String bam, File clientSecret){
```

Comment:
We should switch out strings here for some URI class.  I don't think the existing URI classes support raw paths as a valid input so we probably need to create a wrapper.  We don't want to force people to write `file://somepath` on the command line.